tests/admin_views/tests.py

```diff
@@ -5629,6 +5650,53 @@ def test_url_prefix(self):
                 add_preserved_filters(context, url),
             )
 
+    def test_url_contains_unicode_admin_name(self):
+        # Test URL contains unicode admin name.
+        # For example,
+        # https://SOME_DOMAIN_NAME/後台/User/?is_staff__exact=0&is_superuser__exact=0
+        context = {
+            'preserved_filters': self.get_preserved_filters_querystring(),
+            'opts': User._meta,
+        }
+
+        with override_script_prefix('/後台/'):
+            url = reverse('admin:auth_user_changelist', current_app=self.admin_site.name)
+            self.assertURLEqual(
+                self.get_changelist_url(),
+                add_preserved_filters(context, url),
+            )
+
+    def test_url_unicode_model_name_and_filter_key(self):
+        # Test URL contains unicode model name and unicode filter key.
+        # For example, https://SOME_DOMAIN_NAME/admin/TâiUânPió/?縣市名__exact=0
+        context = {
+            'preserved_filters': self.get_unicode_preserved_filters_querystring(),
+            'opts': TâiUânPió._meta,
+        }
+
+        url = reverse(
+            'admin:admin_views_tâiuânpió_changelist', current_app=self.admin_site.name)
+        self.assertURLEqual(
+            self.get_unicode_changelist_url(),
+            add_preserved_filters(context, url),
+        )
+
+    def test_url_unicode_admin_name_and_model_name_and_filter_key(self):
+        # Test URL contains unicode admin name, model name and unicode filter key.
+        # For example, https://SOME_DOMAIN_NAME/後台/TâiUânPió/?縣市名__exact=0
+        context = {
+            'preserved_filters': self.get_unicode_preserved_filters_querystring(),
+            'opts': TâiUânPió._meta,
+        }
+
+        with override_script_prefix('/後台/'):
+            url = reverse(
```

Comment:
Use a single line. Our [style guide](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/coding-style/#python-style) allows lines up to 119 characters if it helps readability.

Reply:
Thank you, I will reformat it.